ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStoreManager.java

```diff
@@ -0,0 +1,237 @@
+  protected final MetricRegistry metricRegistry;
+  protected final CloudConfig cloudConfig;
+  protected final AzureCloudConfig azureCloudConfig;
+  protected final AzureMetrics azureMetrics;
+  protected final VcrMetrics vcrMetrics;
+  protected final BlobServiceClient azureStorageClient;
+  protected final VerifiableProperties properties;
+  protected final ClusterMap clusterMap;
+
+  /**
+   * Constructor
+   * @param properties Configuration parameters
+   * @param metricRegistry Metrics
+   * @param clusterMap Cluster-map
+   */
+  public AzureStoreManager(VerifiableProperties properties, MetricRegistry metricRegistry, ClusterMap clusterMap) {
+    this.partitionToAzureStore = new ConcurrentHashMap<>();
+    this.metricRegistry = metricRegistry;
+    this.azureMetrics = new AzureMetrics(metricRegistry);
+    this.vcrMetrics = new VcrMetrics(metricRegistry);
+    this.cloudConfig = new CloudConfig(properties);
+    this.azureCloudConfig = new AzureCloudConfig(properties);
+    this.properties = properties;
+    this.clusterMap = clusterMap;
+    this.azureStorageClient =
+        new ConnectionStringBasedStorageClient(cloudConfig, azureCloudConfig, azureMetrics).getStorageSyncClient();
+    testAzureStorageConnectivity();
+    logger.info("Created AzureStoreManager");
+  }
+
+  /**
+   * Tests connection to Azure blob storage
+   */
+  protected void testAzureStorageConnectivity() {
+    PagedIterable<BlobContainerItem> blobContainerItemPagedIterable = azureStorageClient.listBlobContainers();
+    for (BlobContainerItem blobContainerItem : blobContainerItemPagedIterable) {
+      logger.info("blobContainer = {}", blobContainerItem.getName());
+      break;
+    }
+  }
+
+  /**
+   * Returns blob container in Azure for an Ambry partition
+   * @param partitionId Ambry partition
+   * @return blobContainerClient
+   */
+  protected BlobContainerClient getBlobStore(PartitionId partitionId) {
+    BlobContainerClient blobContainerClient;
+    try {
+      blobContainerClient = azureStorageClient.getBlobContainerClient(String.valueOf(partitionId.getId()));
+    } catch (BlobStorageException blobStorageException) {
+      if (blobStorageException.getErrorCode().equals(BlobErrorCode.CONTAINER_NOT_FOUND)) {
+        logger.error("Blob container for partition {} not found due to {}", partitionId.getId(),
+            blobStorageException.getServiceMessage());
+        return null;
+      }
+      vcrMetrics.azureStoreContainerGetError.inc();
+      logger.error("Failed to get blob container for partition {} due to {}", partitionId.getId(),
+          blobStorageException.getServiceMessage());
+      throw blobStorageException;
+    }
+    return blobContainerClient;
+  }
+
+  /**
+   * Creates blob container in Azure for an Ambry partition
+   * @param partitionId Ambry partition
+   * @return blobContainerClient
+   */
+  protected BlobContainerClient createBlobStore(PartitionId partitionId) {
```

Comment:
Why protected?

Reply:
I do not plan on exposing this method. The public API is addBlobStore (called by addReplica) and getBlobStore.